formatter.go

```diff
@@ -27,6 +28,29 @@ const (
 	TimestampMillisFormat = "Jan _2 15:04:05.000"
 )
 
+// LimitByteSlice discards the bytes from a slice that exceeds the limit
+func LimitByteSlice(b []byte, limit int) []byte {
+	if limit > 0 && limit < len(b) {
+		b = append(b[:limit], []byte("...")...)
```

Comment:
I don't think this actually frees the memory until the LogRec is garbage collected, which would be after all the targets have written it out, and that could take some time.   In the case of a string actually exceeding the limit I think it is worth allocating a new byte array that is sized correctly and letting the original be garbage collected.

Reply:
Right.. since we don't grow the slice, we don't actually create a new slice. Good point!